lib/ansible/module_utils/facts.py

```diff
@@ -1208,58 +1213,122 @@ def get_dmi_facts(self):
                 else:
                     self.facts[k] = 'NA'
 
-    @timeout(10)
-    def get_mount_facts(self):
-        uuids = dict()
-        self.facts['mounts'] = []
+    def _run_lsblk(self, lsblk_path):
+        args = ['--list', '--noheadings', '--paths',  '--output', 'NAME,UUID']
+        cmd = [lsblk_path] + args
+        rc, out, err = self.module.run_command(cmd)
+        return rc, out, err
+
+    def _lsblk_uuid(self):
+
+        uuids = {}
+        lsblk_path = self.module.get_bin_path("lsblk")
+        if not lsblk_path:
+            return uuids
+
+        rc, out, err = self._run_lsblk(lsblk_path)
+        if rc != 0:
+            return uuids
+
+        lsblk_lines = out.splitlines()
+
+        # each line will be in format:
+        # <devicename><some whitespace><uuid>
+        # /dev/sda1  32caaec3-ef40-4691-a3b6-438c3f9bc1c0
+        for lsblk_line in lsblk_lines:
+            if not lsblk_line:
+                continue
+
+            line = lsblk_line.strip()
+            fields = line.split()
+            if len(fields) < 2:
+                continue
+
+            device_name, uuid = fields[0].strip(), fields[1].strip()
+            if device_name in uuids:
+                continue
+            uuids[device_name] = uuid
+
+        return uuids
+
+    def _run_findmnt(self, findmnt_path):
+        args = ['--list', '--noheadings', '--notruncate']
+        cmd = [findmnt_path] + args
+        rc, out, err = self.module.run_command(cmd)
+        return rc, out, err
+
+    def _find_bind_mounts(self):
         bind_mounts = []
-        findmntPath = self.module.get_bin_path("findmnt")
-        if findmntPath:
-            rc, out, err = self.module.run_command("%s -lnur" % ( findmntPath ), use_unsafe_shell=True)
-            if rc == 0:
-                # find bind mounts, in case /etc/mtab is a symlink to /proc/mounts
-                for line in out.split('\n'):
-                    fields = line.rstrip('\n').split()
-                    if(len(fields) < 2):
-                        continue
-                    if(re.match(".*\]",fields[1])):
-                        bind_mounts.append(fields[0])
+        findmnt_path = self.module.get_bin_path("findmnt")
+        if not findmnt_path:
+            return
+
+        rc, out, err = self._run_findmnt(findmnt_path)
+        if rc != 0:
+            return bind_mounts
+
+        # find bind mounts, in case /etc/mtab is a symlink to /proc/mounts
+        lines = out.splitlines()
+        for line in lines:
+            fields = line.split()
+            # fields[0] is the TARGET, fields[1] is the SOURCE
+            if len(fields) < 2:
+                continue
+
+            # bind mounts will have a [/directory_name] in the SOURCE column
+            if re.match(".*\]",fields[1]):
```

Comment:
Since we're optimizing to solve this bug, would be good to move regex compilation up to be a one-time thing.  Maybe a class attribute or global for each compiled regex.
``` python

class LinuxHardware(Hardware): 
    BIND_MOUNT_RE = re.compile(r".*\]")
[...]
if re.match(self.BIND_MOUNT_RE, fields[1]):
```